addons/crm/tests/test_crm_pls.py

```diff
@@ -389,3 +390,44 @@ def test_predictive_lead_scoring(self):
         self.assertEqual(tools.float_compare(lead_tag_1.automated_probability, 28.6, 2), 0)
         self.assertEqual(tools.float_compare(lead_tag_2.automated_probability, 28.6, 2), 0)
         self.assertEqual(tools.float_compare(lead_tag_1_2.automated_probability, 28.6, 2), 0)
+
+        # C. Test Update Probabilities Wizard from the settings
+        # Set the PLS config
+        frequency_fields = self.env['crm.lead.scoring.frequency.field'].search([])
+        pls_fields_str = ','.join(frequency_fields.mapped('field_id.name'))
+        pls_start_date_str = "2000-01-01"
+        IrConfigSudo = self.env['ir.config_parameter'].sudo()
+        IrConfigSudo.set_param("crm.pls_start_date", pls_start_date_str)
+        IrConfigSudo.set_param("crm.pls_fields", pls_fields_str)
+
+        # rebuild frequencies table and recompute automated_probability for all leads.
+        Lead._cron_update_automated_probabilities()
+        leads.invalidate_cache()
+
+        self.assertEqual(tools.float_compare(leads[3].automated_probability, 4.21, 2), 0)
+        self.assertEqual(tools.float_compare(leads[8].automated_probability, 0.23, 2), 0)
+
+        # Check that wizard to update lead probabilities has correct value set by default
+        pls_update_wizard = self.env['crm.lead.pls.update'].create({})
+        self.assertEqual(Date.to_string(pls_update_wizard.pls_start_date), pls_start_date_str, 'Correct date is taken from config')
+        self.assertEqual(','.join(pls_update_wizard.pls_fields.mapped('field_id.name')), pls_fields_str, 'Correct fields are taken from config')
+
+        # Update the wizard values and check that config values and probabilities are updated accordingly
+        new_pls_date = "2000-02-02"
+        new_pls_fields = frequency_fields.filtered(lambda f: f.field_id.name in ['source_id', 'lang_id'])
+
+        pls_update_wizard.write({
+            'pls_start_date': Date.from_string(new_pls_date),
+            'pls_fields': new_pls_fields.ids,
+        })
+
+        pls_update_wizard.action_update_crm_lead_probabilities()
+        leads.invalidate_cache()
+
+        # Config params should have been updated
+        self.assertEqual(IrConfigSudo.get_param("crm.pls_start_date"), new_pls_date, 'Correct date should be updated in config')
+        self.assertEqual(IrConfigSudo.get_param("crm.pls_fields"), ','.join(new_pls_fields.mapped('field_id.name')), 'Correct fields should be updated in config')
+
+        # Probabilities should have been updated
+        self.assertEqual(tools.float_compare(leads[3].automated_probability, 91.07, 2), 0)
```

Comment:
I think we could test only that wizard did effectively update parameters. Parameters themselves already have been tested in another unit tests. That way we have each piece tested separately and avoid "big" tests that we cannot update anymore once we forget how it works ...